libcontainer/cgroups/systemd/unified_hierarchy.go

```diff
@@ -136,7 +137,11 @@ func (m *UnifiedManager) Apply(pid int) error {
 	statusChan := make(chan string, 1)
 	if _, err := dbusConnection.StartTransientUnit(unitName, "replace", properties, statusChan); err == nil {
 		select {
-		case <-statusChan:
+		case s := <-statusChan:
+			if s != "done" {
+				dbusConnection.ResetFailedUnit(unitName)
+				return fmt.Errorf("StartTransientUnit job is not done: %s", s)
```

Comment:
1. I would make the error message more user-friendly. Say, something like
```go
"error creating systemd unit `%s`: got `%s`", unitName, s);
```

2. You can use `errors.Errorf()` like everywhere else in this file.